seqr/views/utils/dataset_utils.py

```diff
@@ -94,37 +94,45 @@ def load_mapping_file_content(file_content):
         id_mapping[line[0]] = line[1]
     return id_mapping
 
+
+def _get_individual_sample_lookup(individuals):
+    return {i.individual_id: i for i in individuals}
+
+
+def _get_mapped_individual_lookup_key(sample_id_to_individual_id_mapping):
+    def _get_mapped_id(sample_id):
+        return sample_id_to_individual_id_mapping.get(sample_id, sample_id)
+    return _get_mapped_id
```

Comment:
I like your idea. I may want to do a little further by changing the `return` clause to the following.
```
return _get_mapped_id if sample_id_to_individual_id_mapping else lambda sample_id: sample_id
```
With the change, we can eliminate many `if-else` conditions afterward.